plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/resource/JavaSourceRenameValidator.java

```diff
@@ -0,0 +1,26 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.ext.java.client.resource;
+
+import org.eclipse.che.ide.api.resources.Resource;
+import org.eclipse.che.ide.api.resources.RenamingSupport;
+
+/**
+ * Disable rename operation for the children of the source folder.
+ *
+ * @author Valeriy Svydenko
+ */
+public class JavaSourceRenameValidator implements RenamingSupport {
+    @Override
+    public boolean isRenamed(Resource resource) {
+        return !resource.getParentWithMarker(SourceFolderMarker.ID).isPresent();
```

Comment:
Why do we need this, I think each element which is in the source folder should be renamed by using refactoring, no?